gobblin-iceberg/src/main/java/org/apache/gobblin/iceberg/writer/IcebergMetadataWriter.java

```diff
@@ -729,9 +729,9 @@ private StructLike addLatePartitionValueToIcebergTable(Table table, TableMetadat
   private int isLate(String datepartition, long previousWatermark) {
     ZonedDateTime partitionDateTime = ZonedDateTime.parse(datepartition, HOURLY_DATEPARTITION_FORMAT);
     long partitionEpochTime = partitionDateTime.toInstant().toEpochMilli();
-    if(partitionEpochTime > previousWatermark) {
+    if(partitionEpochTime >= previousWatermark) {
```

Comment:
Why do we change this part? If current watermark is 09am, does that mean data for 09 has completed? And then if we see dat for 09am, is that a late data?

Reply:
if CW = 9am then data for datepartition <= 8am is complete and datepartion>=9 is still not complete (so late =0)